lib/registerTypes/RegisterTypesMapObjects.h

```diff
@@ -55,6 +55,7 @@ void registerTypesMapObjects(Serializer &s)
 		s.template registerType<CGMarket, CGBlackMarket>();
 		s.template registerType<CGMarket, CGUniversity>();
 	s.template registerType<CGObjectInstance, CGHeroPlaceholder>();
+	s.template registerType<IMarket, CGMarket>();
```

Comment:
Can you move this line to the end - to line 135? So that all objects inbetween won't change their order?